src/components/SpectralProfiler/SpectralProfilerComponent.tsx

```diff
@@ -152,25 +152,40 @@ export class SpectralProfilerComponent extends React.Component<WidgetProps> {
     }, 33);
 
-    private genCursoInfoString = (data: Point2D[], smoothedData: Point2D[], cursorXValue: number, cursorXUnit: string, label: string): string => {
-        let cursorInfoString = undefined;
+    private precisionFormatting = (data: number, diff: number, spectralType: SpectralType): string => {
+        return spectralType === SpectralType.CHANNEL ? toFixed(data) : toFormattedNotationByDiff(data, diff);
+    };
+
+    private genCursoInfoString = (data: Point2D[], smoothedData: Point2D[], secondaryXData: number[], cursorXValue: number, cursorXUnit: string, label: string): string => {
+        const frame = this.widgetStore.effectiveFrame;
+
+        let diffLeft: number = undefined;
+        let secondaryXUnit = "";
+        let secondaryChannelString = "";
+        let cursorInfoString: string = "";
+
         const nearest = binarySearchByX(data, cursorXValue);
         const nearestSmooth = smoothedData.length ? binarySearchByX(smoothedData, cursorXValue) : null;
         if (nearest?.point && nearest?.index >= 0 && nearest?.index < data?.length) {
-            let floatXStr = "";
+            let primaryXStr: string = "";
             let smoothedFloatXStr = "";
-            const diffLeft = nearest.index - 1 >= 0 ? Math.abs(nearest.point.x - data[nearest.index - 1].x) : 0;
-            if (diffLeft > 0 && diffLeft < 1e-6) {
-                floatXStr = formattedNotation(nearest.point.x);
-                smoothedFloatXStr = formattedNotation(nearestSmooth.point.x);
-            } else if (diffLeft >= 1e-6 && diffLeft < 1e-3) {
-                floatXStr = toFixed(nearest.point.x, 6);
-                smoothedFloatXStr = toFixed(nearestSmooth?.point.x, 6);
-            } else {
-                floatXStr = toFixed(nearest.point.x, 3);
-                smoothedFloatXStr = toFixed(nearestSmooth?.point.x, 3);
+            const currentIndex = nearest.index;
+            const neighborIndex = currentIndex > 0 ? currentIndex - 1 : currentIndex + 1;
+            diffLeft = data.length === 1 ? 1e-9 : Math.abs(data[currentIndex].x - data[neighborIndex].x);
+            primaryXStr = this.precisionFormatting(data[currentIndex].x, diffLeft, frame.spectralType);
+            smoothedFloatXStr = this.precisionFormatting(nearestSmooth?.point.x, diffLeft, frame.spectralType);
+
+            let xLabel = cursorXUnit === "Channel" ? `Channel ${primaryXStr}` : `${primaryXStr}${cursorXUnit ? ` ${cursorXUnit}` : ""}`;
+
+            if (this.widgetStore.secondaryAxisCursorInfoVisible) {
+                diffLeft = data.length === 1 ? 1e-9 : Math.abs(secondaryXData[currentIndex] - secondaryXData[neighborIndex]);
+                const secondaryXStr = this.precisionFormatting(secondaryXData[currentIndex], diffLeft, frame.spectralTypeSecondary);
+
+                if (frame.spectralTypeSecondary !== SpectralType.CHANNEL) secondaryXUnit = frame.spectralUnitSecondary;
+                else secondaryChannelString = "Channel";
```

Comment:
Angus has told me in my previous PR that he wants to always have brackets for if-statements.

Reply:
fixed